### PR TITLE
modern python

### LINKED ISSUE #70: Enable postprocess on dag

### LINKED ISSUE #71: Pass transaction and shard hints into pipeline when set.

### PR DESCRIPTION
**Pull request by chayim**
Modern build system for python components. This includes linters, security scanning, and parallel builds across supported versions.

Also, by fixing linter errors.

closes #70 
closes #71

### CODE REVIEW

**Comment by chayim** [comment]
@DvirDukhan Can you modify the merge requirements so that this depends on ci/circleci:build-latest rather that build?

**Comment by chayim** [comment]
@DvirDukhan This is ready, we've merged in the fix. Can you (please) change the repo requirement to no longer be dependent on ci/circleci:build and instead require ci/circleci:build-latest, as part of this.

While I can slam it in - I'd like a review.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ARG OSNICK=bionic
```
[the remark this answers is not in the document]
Comment (chayim) [reply Dockerfile:1]: A future goal - I'd like to build dockers out of these, and given people a one stop shop. We are not currently using it.

File: redisai/__init__.py [hunk 2]
```diff
@@ -1,3 +1,3 @@
-from .client import Client
+from .client import Client  # noqa
```
[the remark this answers is not in the document]
Comment (chayim) [reply redisai/__init__.py:1]: #noqa tells the python linter to not ignore this specific line. It sets off a linter error but is required.

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,97 +1,139 @@
...
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "test-requirements.txt" }}
+            if [[ $CIRCLE_BRANCH == *docs ]]; then
+              echo "Identifies as documents PR, no testing required"
+              circleci step halt
+            fi
 
+  abort_for_noci:
```
[the remark this answers is not in the document]
Comment (chayim) [reply .circleci/config.yml:15]: This is the generic name currently in use in all clients.

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**